File: frontend/src/services/fertilizerApplicationService.ts
import { FertilizerApplication, FertilizerApplicationRequest } from '../types/AgronomicTypes'
import { api } from './api'

export const fertilizerApplicationService = {
  async getByCropHistory(cropHistoryId: number): Promise<FertilizerApplication[]> {
    const response = await api.get<FertilizerApplication[]>('/fertilizer-applications', {
      params: { cropHistoryId },
    })
    return response.data
  },

  async getById(id: number): Promise<FertilizerApplication> {
    const response = await api.get<FertilizerApplication>(`/fertilizer-applications/${id}`)
    return response.data
  },

  async create(data: FertilizerApplicationRequest): Promise<FertilizerApplication> {
    const response = await api.post<FertilizerApplication>('/fertilizer-applications', data)
    return response.data
  },

  async update(id: number, data: FertilizerApplicationRequest): Promise<FertilizerApplication> {
    const response = await api.put<FertilizerApplication>(`/fertilizer-applications/${id}`, data)
    return response.data
  },

  async delete(id: number): Promise<void> {
    await api.delete(`/fertilizer-applications/${id}`)
  },
}

File: agriculture-field-service/src/main/java/com/omstu/agriculturefield/crop/service/FertilizerApplicationService.java
package com.omstu.agriculturefield.crop.service;

import com.omstu.agriculturefield.crop.dto.FertilizerApplicationRequest;
import com.omstu.agriculturefield.crop.dto.FertilizerApplicationResponse;

import java.util.List;

public interface FertilizerApplicationService {
    List<FertilizerApplicationResponse> getByCropHistoryId(Long cropHistoryId);
    FertilizerApplicationResponse getById(Long id);
    FertilizerApplicationResponse create(FertilizerApplicationRequest request);
    FertilizerApplicationResponse update(Long id, FertilizerApplicationRequest request);
    void delete(Long id);
}
